torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -478,15 +480,43 @@ void IRPrinter::handle(const ReductionOp* rop) {
 }
 
 void IRPrinter::handle(const BroadcastOp* bop) {
-  indent();
-  handle(bop->out());
-  os << "\n";
-  indent_size++;
-  indent();
-  os << " = ";
-  handle(bop->in());
-  indent_size--;
-  os << ";\n";
+  const ir_utils::ParallelTypeBitmap domains =
+      ir_utils::getParallelBroadcastDomains(bop, getThreadPredicateMap());
+  const bool thread_x = domains.get(ParallelType::TIDx);
+  const bool thread_y = domains.get(ParallelType::TIDy);
+  const bool thread_z = domains.get(ParallelType::TIDz);
+  const bool block_x = domains.get(ParallelType::BIDx);
+  const bool block_y = domains.get(ParallelType::BIDy);
+  const bool block_z = domains.get(ParallelType::BIDz);
+
+  const bool grid_broadcast_needed = block_x || block_y || block_z;
+  const bool block_broadcast_needed = thread_x || thread_y || thread_z;
+
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Do we also have this check in `IterDomain::parallelize`?

Reply:
Yes. It's checked in `ir_utils::getParallelBroadcastDomains`. https://github.com/naoyam/pytorch/blob/block_broadcast/torch/csrc/jit/codegen/cuda/lower_utils.cpp#L596